### PR TITLE
[fuchsia] Make it easy to flip the whole system between flutter_jit_a…

### PR DESCRIPTION
**Pull request by rmacnak-google**
…pp and flutter_aot_app.

### CODE REVIEW

**Comment by krisgiesing** [comment]
That was fast :)

So this does #1 and #2 from the list in the email thread, yes? And #3 would presumable be a follow on change?

**Comment by rmacnak-google** [comment]
Right. As for 3), I'm open to name suggestions for a GN variable :)

File: examples/hello_flutter/BUILD.gn [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import("//flutter/build/flutter_app.gni")
 
-flutter_app("hello_flutter") {
+flutter_jit_app("hello_flutter") {
```
Comment (krisgiesing) [inline examples/hello_flutter/BUILD.gn]: Does this need to change? I thought most clients would use flutter_app and follow external configuration?

Comment (zanderso) [inline examples/hello_flutter/BUILD.gn]: I think the idea here is to build the hello_flutter app both ways to make sure that both ways work.

File: build/flutter_app.gni [hunk 2]
```diff
@@ -374,3 +348,59 @@ template("flutter_aot_app") {
...
+#
+#   analysis_options (optional)
+#     By default, a script to run the analyzer on the contents of the package is
+#     generated in the output directory. This parameter contains the path to an
+#     analysis options file for this target.
+#
+#   disable_analysis (optional)
+#     Prevents the analysis script from being generated.
+template("flutter_app") {
+  if (true) {
```
Comment (zanderso) [inline build/flutter_app.gni]: I'm not sure I like the idea of this being toggled by a global flag, but I think it's good to be explicit that there's a choice here.

File: build/flutter_app.gni [hunk 3]
```diff
@@ -18,33 +18,7 @@ flutter_tools_label = "$flutter_root/packages/flutter_tools:fuchsia_builder($hos
...
-#
-#   analysis_options (optional)
-#     By default, a script to run the analyzer on the contents of the package is
-#     generated in the output directory. This parameter contains the path to an
-#     analysis options file for this target.
-#
-#   disable_analysis (optional)
-#     Prevents the analysis script from being generated.
-template("flutter_app") {
+template("flutter_jit_app") {
```
Comment (zanderso) [inline build/flutter_app.gni]: Maybe "flutter_dynamic_app" or "dynamic_flutter_app"

Comment (zanderso) [inline build/flutter_app.gni]: And "flutter_static_app" or "static_flutter_app" below.

File: examples/hello_flutter/BUILD.gn (hunk 1 above)
Comment (rmacnak-google) [inline examples/hello_flutter/BUILD.gn]: The intent is to ensure there's at least one app built both ways. There's a flutter_aot_app just below here.

**Pull request merged**